standalone/extHistoryEnumerator.py

```diff
@@ -0,0 +1,66 @@
+"""
+This script enumerates all occurrences of a text, based on a regex.
+The following search example returns all soapaction names found in 
+    the history, be it in the header or body.
+"""
+
+from org.parosproxy.paros.control import Control;
+from org.parosproxy.paros.extension.history import ExtensionHistory;
+from org.parosproxy.paros.model import Model;
+from org.parosproxy.paros.model import HistoryReference;
+import re;
+from org.parosproxy.paros.view import AbstractFrame;
+from javax.swing import JPanel;
+from org.zaproxy.zap.utils import ZapTextArea;
+
+sessionId = Model.getSingleton().getSession();
+tbHist = Model.getSingleton().getDb().getTableHistory();
+regex='soapaction';
+
+def search(msg):
+  match = re.search(regex, msg.getRequestHeader().toString(), re.IGNORECASE);
+  if(match):
+    for header in msg.getRequestHeader().toString().split("\n"):
+      match = re.search(regex, header, re.IGNORECASE);
+      if(match):
+        yield header;
+  match = re.search(regex, msg.getRequestBody().toString(), re.IGNORECASE);
+  if(match):
+    yield match;
+  match = re.search(regex, msg.getResponseHeader().toString(), re.IGNORECASE);
```

Comment:
Might be worth commenting here `Search Response`

Just so it sticks out for other users, and doesn't seem like duplicate code.